DOC: update the Series.str.ismethods docstring

MrKriss opened the pull request:
- [ ] closes #xxxx
- [x] tests ~added~ / passed (using scripts/validate_docstrings.py)
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

A result of the Pandas Sprint at PyData London 2018.

Commit:
Enhance docstrings for Series.str.ismethods

datapythonista commented on pandas/core/strings.py:
```diff
@@ -2401,11 +2401,137 @@ def rindex(self, sub, start=0, end=None):
+    **Checks for Whitespace**
+
+    >>> # All characters represent whitespace
+    >>> s4 = pd.Series([' ','\\t\\r\\n ', ''])
```
I'd find a bit clearer to use `r'\t\r\n'` than `'\\t\\r\\n'`

MrKriss replied:
So using `r'\t\r\n'` seems to interfere with the validation script causing an error, likely because the newline and whitespace characters are still rendered as the docstring is parsed. I can correct for this if I make the whole docstring a raw string, but that seems a bit drastic, and not sure if it would have any other consequences. Would sticking with '\\t\\r\\n' be preferable over this?

datapythonista commented on pandas/core/strings.py:
```diff
@@ -2401,11 +2401,137 @@ def rindex(self, sub, start=0, end=None):
+
+    >>> s1 = pd.Series(['AB', 'C12', '42', ''])
+
+    >>> # All are alphabetic characters
```
I find these comments before each function unnecessary.

MrKriss replied:
Sure, I can strip them out where it's more self-explanatory, though I'm thinking to keep the ones in the section on *More Detailed Checks for Numeric Characters* as I found this the most confusing part. i.e. why there are so many checks for numeric values and what the difference between them is. Or I could put these as text instead of comments.

datapythonista replied:
Sounds good to me, the ones that you think the example is not clear by itself I think it's better to have the explanations as text and not as code comments.

datapythonista commented on pandas/core/strings.py:
```diff
@@ -2401,11 +2401,137 @@ def rindex(self, sub, start=0, end=None):
+    --------
+    **Checks for Alphabetic and Numeric Characters**
+
+    >>> s1 = pd.Series(['AB', 'C12', '42', ''])
```
We haven't done it anywhere yet, but I think it could make sense to add an index with the same values. It'd make very easy to follow the examples.

As a personal opinion, I find a bit distracting using too arbitrary examples. I'd prefer a real-world example, but as seems difficult for this case, I'd prefer something like `['one', 'one1', '1', '']`, which makes it obvious what is being shown, and don't let the users guessing why `42` and not `43`.

MrKriss replied:
Just checking I understand, do you mean having the series values and index be the same so the values are displayed side by side like below?  

```
>>> s1 = pd.Series(data=['one', 'one1', '1', ''], 
                   index=['one', 'one1', '1', ''])

>>> s1.str.isalpha()
one      True
one1    False
1       False
        False
dtype: bool
```

datapythonista replied:
It's just an idea, but yes, that's what I meant. We haven't done it in any docstring yet, afaik. But I think it makes very easy to see which value is true for each method.

MrKriss replied:
It does make comparisons easier, though I find having a blank in the last index position spoils it a bit. Could explicitly label it with 'empty string' as below?
```
>>> s1.str.isalpha()
one              True
one1            False
1               False
empty string    False
dtype: bool
```

datapythonista reviewed (commented):
Really good job, added some comments with ideas, and about the standards followed.

Commit:
Specify file encoding to fix python 2.7 tests/checks

Commit:
Ammend docstring following feedback

- Made docstring header fit on one line
- Expanded return value dtype
- Switched explanations in comments to text
- Simplified examples
- Fixed typos
